tests/tracer/test_logging.py

```diff
@@ -0,0 +1,356 @@
+
+ddtrace_logger.warning('warning log')
+"""
+
+    out, err, status, pid = ddtrace_run_python_code_in_subprocess(code)
+    assert status == 0, err
+    assert re.search(LOG_PATTERN, str(err)) is not None
+    assert b"warning log" in err
+    assert out == b""
+
+
+def test_warn_logs_can_go_to_file(run_python_code_in_subprocess, ddtrace_run_python_code_in_subprocess, tmpdir):
+    """
+    When setting up the default ddtrace logger when not in debug mode,
+        it can log to a file at WARN log level.
+    """
+    env = os.environ.copy()
+    log_file = tmpdir + "/testlog.log"
+    env["DD_TRACE_LOG_FILE"] = log_file
+    env["DD_TRACE_FILE_SIZE_BYTES"] = "200000"
+    code = """
+import logging
+import ddtrace
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.WARN
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.handlers.RotatingFileHandler)
+assert ddtrace_logger.handlers[0].maxBytes == 200000
+assert ddtrace_logger.handlers[0].backupCount == 1
+
+ddtrace_logger.warning('warning log')
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err == b""
+    assert out == b""
+    with open(log_file) as file:
+        content = file.read()
+        assert len(content) > 0
+        assert "warning log" in content
+        assert re.search(LOG_PATTERN, content) is not None
+
+    code = """
+import logging
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.WARN
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.handlers.RotatingFileHandler)
+assert ddtrace_logger.handlers[0].maxBytes == 200000
+assert ddtrace_logger.handlers[0].backupCount == 1
+
+ddtrace_logger.warning('warning log')
+"""
+
+    out, err, status, pid = ddtrace_run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert err == b""
+    assert out == b""
+    with open(log_file) as file:
+        content = file.read()
+        assert len(content) > 0
+        assert "warning log" in content
+        assert re.search(LOG_PATTERN, content) is not None
+
+
+def test_debug_logs_go_to_stderr(run_python_code_in_subprocess, ddtrace_run_python_code_in_subprocess):
+    """
+    When setting up the default logger in debug mode without a log path,
+        it automatically logs to stderr.
+    """
+    env = os.environ.copy()
+    env["DD_TRACE_DEBUG"] = "true"
+    code = """
+import logging
+import ddtrace
+
+ddtrace_logger = logging.getLogger('ddtrace')
+assert ddtrace_logger.level == logging.DEBUG
+assert len(ddtrace_logger.handlers) == 1
+assert isinstance(ddtrace_logger.handlers[0], logging.StreamHandler)
+"""
+
+    out, err, status, pid = run_python_code_in_subprocess(code, env=env)
+    assert status == 0, err
+    assert re.search(LOG_PATTERN, str(err)) is not None
+    assert b"debug logs" in err
```

Comment:
this is asserting on the startup debug logs correct? since I don't see an explicit log above.

Reply:
Resolving this since this assertion doesn't exist.